userspace/libsinsp/user.cpp

```diff
@@ -495,3 +513,45 @@ void sinsp_usergroup_manager::notify_group_changed(const scap_groupinfo *group,
 	m_inspector->m_pending_state_evts.push(cevt);
 #endif
 }
+
+void sinsp_usergroup_manager::load_from_container(const std::string &container_id, const std::string &overlayfs_root)
+{
+	if (!m_import_users)
+	{
+		return;
+	}
+
+	if (overlayfs_root.empty())
+	{
+		// Avoid loading from host
+		return;
+	}
+
+	// See fgetpwent() feature test macros: https://man7.org/linux/man-pages/man3/fgetpwent.3.html
+#if defined HAVE_PWD_H && _DEFAULT_SOURCE
+	auto passwd_in_container = overlayfs_root + "/etc/passwd";
+	auto pwd_file = fopen(passwd_in_container.c_str(), "r");
+	if(pwd_file)
+	{
+		while(auto p = fgetpwent(pwd_file))
+		{
+			m_inspector->m_usergroup_manager.add_user(container_id, p->pw_uid, p->pw_gid, p->pw_name, p->pw_dir, p->pw_shell, true);
+		}
+		fclose(pwd_file);
+	}
+#endif
+
+	// See fgetgrent() feature test macros: https://man7.org/linux/man-pages/man3/fgetgrent.3.html
+#if defined HAVE_GRP_H && _DEFAULT_SOURCE
```

Comment:
Leaving a note for myself:
```suggestion
#if defined HAVE_GRP_H && (defined _DEFAULT_SOURCE || defined _SVID_SOURCE)
```